// wafer-node-sdk/lib/imgUrlData/index.js
var http = require('http');
var mysql = require('mysql');
var url = require('url');
var querystring = require('querystring');


//创建数据库连接
var connection = mysql.createConnection({
  host: '10.66.194.126',
  user: 'root',
  password: '13545715089sq',
  database: 'cAuth',
  port: '3306'
});
//连接mysql
connection.connect(function (err) {
  if (err) {
    //数据库连接失败，并打印错误信息
    console.log('query-' + err);
    return;
  }
  //数据库连接成功
  console.log('conection connect succedd');
})






function img (id,urls) {

//nodejs服务器处理meet.js页面wx.request上来的数据
var server = http.createServer(function (req, res) {
  var url = url.parse(req.url, true);
  var data = querystring.stringify(url.query);
  //输出url_info
  console.log("url",data);

  //解析request过来的参数
  var param = req.query;
  if (param) {
    var id = url.parse(decodeURI(request.url), true).query.id; //解析参数为id的值  
    if (id)
      console.log(id);
    var urls = url.parse(decodeURI(request.url), true).query.urls;//解析参数为urls的值  
    if (urls)
      console.log(urls);

    //MySQL数据库写入数据方法，将id和urls值写入photolist表
    var userAddsql = 'insert into photolist(id,url) values(?,?)';
    var param = [id, urls];

    connection.query(userAddsql, param, function (err, rs) {
      if (err) {
        //打印错误信息
        console.log('insert err', err.message);
        return;
      }
      //打印数据写入成功
      console.log('insert success');
    });

    var obj = new Object();

    //关闭connect连接
    connection.end(function (err) {
      if (err) {
        return;
      }
      //打印关闭数据库成功
      console.log('connection end succedd');
    })
  }
    
});
//服务器监听1337端口
server.listen(1337, '127.0.0.1');


//打印服务器监听端口
console.log('listening on port  1337');

}


module.exports = img
